apps/src/codebridge/Settings/SettingsDropdown.tsx

```diff
@@ -0,0 +1,140 @@
+import {trySetSessionStorage} from '@cdo/apps/utils';
+import commonI18n from '@cdo/locale';
+
+import {useCodebridgeContext} from '../codebridgeContext';
+
+import moduleStyles from './settings-dropdown.module.scss';
+
+// fontSizeOptions contains a list of value/text from the FontSize enum,
+// e.g., [{value: 'Tiny', text: 'Tiny'}, {value: 'Small', text: 'Small'}, ...]
+const fontSizeOptions: SimpleDropdownProps['items'] = Object.keys(FontSize)
+  .filter(key => isNaN(Number(key))) // Filters out the reverse enum keys.
+  .map(key => ({
+    value: key,
+    text: key,
+  }));
+interface SettingsDropdownProps {
+  closeDropdown: () => void;
+  buttonRef: React.RefObject<HTMLDivElement>;
+}
+
+const SettingsDropdown: React.FunctionComponent<SettingsDropdownProps> = ({
+  closeDropdown,
+  buttonRef,
+}) => {
+  const menuRef = useOutsideClick<HTMLDivElement>(closeDropdown);
+  const editorFontSizeKey = useAppSelector(
+    state => state.lab2View.editorFontSizeKey
+  );
+  const selectedFontSizeKey = editorFontSizeKey || DEFAULT_FONT_SIZE_KEY;
+  const {signInState} = useAppSelector(state => state.currentUser);
+  const {levelProperties} = useCodebridgeContext();
+  const dispatch = useAppDispatch();
+  const [selectedValue, setSelectedValue] = useState(selectedFontSizeKey);
+
+  const getSelectedKey = (value: string) => value as keyof typeof FontSize;
+
+  const dropdownStyles = useDropdownPosition(buttonRef, menuRef);
+
+  const onTextEditorDropdownChange = async (value: string) => {
+    setSelectedValue(getSelectedKey(value));
+  };
+
+  const onSave = () => {
+    const selectedKey = getSelectedKey(selectedValue);
+    if (FontSize[selectedKey]) {
+      // We want the user preference for selected font size to persist across a session
+      // for signed-in users per app type.
+      if (signInState === SignInState.SignedIn) {
+        const sessionStorageKey = `${levelProperties.appName}CodeEditorFontSizeKey`;
+        trySetSessionStorage(sessionStorageKey, selectedKey);
+      }
+      dispatch(setEditorFontSize(selectedKey));
+    }
+    closeDropdown();
+  };
+
+  return createPortal(
+    <FocusTrap
+      focusTrapOptions={{
+        onDeactivate: closeDropdown,
+        clickOutsideDeactivates: true,
+      }}
+    >
+      <div
+        className={moduleStyles.settingsDropdown}
+        ref={menuRef}
+        role="dialog"
+        style={dropdownStyles}
+        aria-modal="true"
+        aria-label={commonI18n.settings()}
+      >
+        <div className={moduleStyles.header}>
+          <Heading6 className={moduleStyles.settingsTitle}>
+            {commonI18n.settings()}
+          </Heading6>
+          <CloseButton
+            onClick={closeDropdown}
+            aria-label="Close settings"
+            id="close-settings-dropdown"
+          />
+        </div>
+        <div className={moduleStyles.dropdownRow}>
+          {/* Customized label for dropdown */}
```

Comment:
@hannahbergam - would you mind taking a look at this section with a11y in mind? Thank you!

Reply:
Loving the focus trap and the labels! I can do this branch with my screenreader if that's what you have in mind?

Reply:
Yes thank you! Just want to make sure I am following best a11y practices. Appreciate you!